src/main/resources/api/service_openapi.yaml

```diff
@@ -1153,6 +1194,192 @@ components:
         memberEmail:
           type: string
 
+    CreateControlledGoogleBucketRequestBody:
+      description: Payload for requesting a new controlled resource.
+      type: object
+      properties:
+        common:
+          $ref: '#/components/schemas/ControlledResourceCommonFields'
+        googleBucket:
+          $ref: '#/components/schemas/GoogleBucketCreationParameters'
+
+    ControlledResourceCommonFields:
+      type: object
+      required: [ name, cloningInstructions, jobControl, accessScope ]
+      properties:
+        jobControl:
+          $ref: '#/components/schemas/JobControl'
+        name:
+          $ref: "#/components/schemas/Name"
+        description:
+          type: string
+        cloningInstructions:
+          $ref: '#/components/schemas/CloningInstructionsEnum'
+        accessScope:
+          type: string
+          enum: [ 'SHARED', 'PRIVATE' ]
+        managedBy:
+          type: string
+          enum: [ 'USER', 'APPLICATION' ]
+        privateResourceUser:
+          $ref: '#/components/schemas/PrivateResourceUser'
+        visible:
+          type: boolean
+        owner:
+          type: string
+
+    PrivateResourceUser:
+      description: A user and role pair for a user of a private resource.
+      type: object
+      properties:
+        userName:
+          description: Email address for the user of the resource
+          type: string
+        iamRole:
+          $ref: '#/components/schemas/IamRole'
+
+    CreatedControlledGoogleBucket:
+      description: Response Payload for requesting a new controlled resource.
+      type: object
+      properties:
+        resourceId:
+          description: UUID of a newly-created resource. Null if not created yet.
+          type: string
+          format: uuid
+        jobReport:
+          $ref: '#/components/schemas/JobReport'
+        googleBucket:
+          $ref: '#/components/schemas/GoogleBucketStoredAttributes'
+        errorReport:
+          $ref: '#/components/schemas/ErrorReport'
+
+    GoogleBucketCreationParameters:
+      description: >-
+        Bucket-specific properties to be set on creation. These are a subset of the
+        values accepted by the Google Storage API.
+      type: object
+      properties:
+        name:
+          description: A valid bucket name per https://cloud.google.com/storage/docs/naming-buckets.
+          type: string
+        location:
+          description: A valid bucket location per https://cloud.google.com/storage/docs/locations.
+          type: string
+        defaultStorageClass:
+          $ref: '#/components/schemas/GoogleBucketDefaultStorageClass'
+        lifecycle:
+          $ref: '#/components/schemas/GoogleBucketLifecycle'
+    GoogleBucketStoredAttributes: # todo: use bucket UID type here
```

Comment:
is this todo tracked in a ticket?

Reply:
I don't think I want to do this anymore. It locks us into only exposing the fields on that (unrelated) type, currently just name, or breaking the API to add another.

Reply:
Agreed that replacing `GoogleBucketStoredAttributes` with the UID makes for a fragile interface, but you could make the `bucketName` property a `GoogleBucketUid` object instead of a string

Reply:
That doesn't buy me any type safety or useful abstraction, and just puts the bucket name one level down under a nonstandard "uid" property. I don't feel like it's enough to warrant reuse on its own.